scripts/language_model/run_glue.py

```diff
@@ -0,0 +1,603 @@
+parser.add_argument(
+    '--log_interval',
+    type=int,
+    default=10,
+    help='report interval')
+parser.add_argument(
+    '--max_len',
+    type=int,
+    default=128,
+    help='Maximum length of the sentence pairs')
+
+parser.add_argument(
+    '--pad',
+    default=True,
+    action='store_true',
+    help='Whether to pad to maximum length when preparing data batches. '
+         'Have to be true currently due to left padding')
+
+parser.add_argument(
+    '--seed', type=int, default=2, help='Random seed')
+
+parser.add_argument(
+    '--accumulate',
+    type=int,
+    default=None,
+    help='The number of batches for gradients accumulation to simulate large batch size. '
+         'Default is None')
+parser.add_argument(
+    '--gpu', type=int, default=None, help='Number of gpus for finetuning.')
+parser.add_argument(
+    '--cpu', type=int, default=None, help='Number of cpus for finetuning.')
+parser.add_argument(
+    '--task_name',
+    default='MRPC',
+    type=str,
+    help='The name of the task to fine-tune.')
+
+parser.add_argument(
+    '--model',
+    type=str,
+    default='xlnet_cased_l12_h768_a12',
+    help='The name of pre-trained XLNet model to fine-tune')
+
+parser.add_argument(
+    '--dataset',
+    type=str,
+    default='126gb',
+    help='The dataset BERT pre-trained with.')
+
+parser.add_argument(
+    '--output_dir',
+    type=str,
+    default='./output_dir',
+    help='The output directory where the model params will be written.')
+
+parser.add_argument(
+    '--only_inference',
+    action='store_true',
+    help='If set, we skip training and only perform inference on dev and test data.')
+
+parser.add_argument(
+    '--dtype',
+    type=str,
+    default='float32',
+    choices=['float32', 'float16'],
+    help='The data type for training. Doesnt support float16 currently')
+
+parser.add_argument(
+    '--model_parameters',
+    type=str,
+    default=None,
+    help='A parameter file for the model that is loaded into the model'
+    ' before training/inference. It is different from the parameter'
+    ' file written after the model is trained.')
+
+parser.add_argument(
+    '--early_stop',
+    type=int,
+    default=None,
+    help='Whether to perform early stopping based on the metric on dev set. '
+         'The provided value is the patience. ')
+
+
+args = parser.parse_args()
+
+
+
+def split_and_load(arrs, ctx):
+    """split and load arrays to a list of contexts"""
+    assert isinstance(arrs, (list, tuple))
+    # split and load
+    loaded_arrs = [mx.gluon.utils.split_and_load(arr, ctx, even_split=False) for arr in arrs]
+    return zip(*loaded_arrs)
+
+
+logging.getLogger().setLevel(logging.INFO)
+logging.captureWarnings(True)
+logging.info(args)
+
+batch_size = args.batch_size
+dev_batch_size = args.dev_batch_size
+task_name = args.task_name
+lr = args.lr
+epsilon = args.epsilon
+accumulate = args.accumulate
+log_interval = args.log_interval * accumulate if accumulate else args.log_interval
```

Comment:
It seems not necessary to introduce an extra name for each of these. In fact in the code below you don't consistently use the either version. Let's just use the `args.X` version instead of `X` and delete these lines? Same for `bert/run_glue.py`